### PR TITLE
refactoring matmul_v2 mkldnn hierarchy

### PR DESCRIPTION
**Pull request by sfraczek**
### PR types
Others

### PR changes
OPs

### Describe
Refactoring of mkldnn matmul v2. Using composition of matmul-grad in matmul_v2-grad instead of inheritance.
Instead of chain of inheritance have operators inherit only from paddle::framework::OpKernel<T> as done in other operators.
Some functions have been extracted from classes to allow code reuse without inheritance.
Alternative idea I have is to replace entire class declaration (MatMulGradMKLDNNKernel) in matmul_mkldnn_op.h with declaration of a single method such as MatmulV1GradMKLDNNCompute() and extract necessary methods from the class to do that.

### CODE REVIEW

**Review by jczaja — DISMISSED** [review]
LGTM

**Review by lidanqing-vv — DISMISSED** [review]
LGTM

File: paddle/fluid/operators/mkldnn/matmul_v2_mkldnn_op.cc [hunk 1]
```diff
@@ -123,45 +123,62 @@ class MatMulV2MKLDNNHandler
...
+  return !fused_reshape_Out.empty() && !fused_transpose_Out.empty();
+}
+
+float ComputeOutputScale(const ExecutionContext& ctx) {
+  float scale_x = ctx.Attr<float>("Scale_x");
+  float scale_y = ctx.Attr<float>("Scale_y");
+  bool force_fp32_out = ctx.Attr<bool>("force_fp32_output");
+  float scale_out = force_fp32_out ? 1.f : ctx.Attr<float>("Scale_out");
+  if (scale_x != 1 || scale_y != 1 || force_fp32_out != false) {
+    std::cout << "scale_x: " << scale_x << ", scale_y: " << scale_y
```
Comment (wozna) [inline paddle/fluid/operators/mkldnn/matmul_v2_mkldnn_op.cc]: I think it is better to use `LOG(INFO)` to have control of the output

**Comment by sfraczek** [comment]
Please merge, this change is ready.

File: paddle/fluid/operators/mkldnn/matmul_v2_mkldnn_op.cc (hunk 1 above)
Comment (sfraczek) [reply paddle/fluid/operators/mkldnn/matmul_v2_mkldnn_op.cc]: Oh actually I forgot to delete this! thanks for noticing

**Comment by sfraczek** [comment]
@jczaja I believe it's ready for merging

**Review by wozna — APPROVED** [review]
LGTM

**Review by jczaja — APPROVED** [review]
LGTM

**Pull request merged**